tests/components/header_footer_template.test.jsx

```diff
@@ -8,13 +8,30 @@ import {shallow} from 'enzyme';
 import NotLoggedIn from 'components/header_footer_template/header_footer_template.jsx';
 
 describe('components/HeaderFooterTemplate', () => {
+    const RealDate = Date;
+
+    function mockDate(date) {
+        global.Date = class extends RealDate {
+            constructor() {
+                super();
+                return new RealDate(date);
+            }
+        };
+    }
```

Comment:
Here's is the actual fix for currentYear snapshots failure.

Others are just small cleanup/updates.